packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -331,4 +271,71 @@ export class QuoteRequestor {
         const currentTimeMs = new BigNumber(Date.now());
         return expirationTimeMs.isLessThan(currentTimeMs.plus(this._expiryBufferMs));
     }
+
+    private async _getQuotesAsync<ResponseT>(
+        makerAssetData: string,
+        takerAssetData: string,
+        assetFillAmount: BigNumber,
+        marketOperation: MarketOperation,
+        options: RfqtRequestOpts,
+        quoteType: 'firm' | 'indicative',
+    ): Promise<ResponseT[]> {
+        // create an array of promises for quote responses, using "undefined"
+        // as a placeholder for failed requests.
+        const responsesIfDefined: Array<undefined | AxiosResponse<ResponseT>> = await Promise.all(
+            Object.keys(this._rfqtAssetOfferings).map(async url => {
+                if (this._makerSupportsPair(url, makerAssetData, takerAssetData)) {
+                    let logEntry;
+                    const timeBeforeAwait = Date.now();
+                    try {
+                        const quotePath = (() => {
+                            switch (quoteType) {
+                                case 'firm':
+                                    return 'quote';
+                                    break;
+                                case 'indicative':
+                                    return 'price';
+                                    break;
+                                default:
+                                    throw new Error(`Unexpected quote type ${quoteType}`);
+                            }
+                        })();
+                        const response = await Axios.get<ResponseT>(`${url}/${quotePath}`, {
+                            headers: { '0x-api-key': options.apiKey },
+                            params: {
+                                takerAddress: options.takerAddress,
+                                ...inferQueryParams(marketOperation, makerAssetData, takerAssetData, assetFillAmount),
+                            },
+                            timeout: options.makerEndpointMaxResponseTimeMs,
+                        });
+                        logEntry = {
+                            statusCode: response.status,
```

Comment:
to we want to include `quoteType` here now?

Reply:
Yup, just added that in a commit that came in I guess right after your review.  See here: https://github.com/0xProject/0x-monorepo/pull/2581/commits/c929782e0d087f2051e673d20c672478807b8dcd#diff-951722c8cfc47f2ea3eebf81342ae48dR292